ngraph/core/src/op/einsum.cpp

```diff
@@ -0,0 +1,278 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <algorithm>
+#include <cctype>
+#include <ngraph/validation_util.hpp>
+#include <regex>
+#include <string>
+#include <unordered_map>
+
+#include "itt.hpp"
+#include "ngraph/op/einsum.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v7::Einsum, "Einsum", 7);
+
+op::v7::Einsum::Einsum(const OutputVector& inputs, const std::string& equation)
+    : Op(inputs)
+    , m_equation(equation)
+{
+    constructor_validate_and_infer_types();
+}
+
+bool op::v7::Einsum::parse_equation(std::vector<std::string>& input_subscripts,
+                                    std::string& output_subscript)
+{
+    NGRAPH_OP_SCOPE(v7_Einsum_parse_equation);
+
+    // create regular expression to match subscript that consists of either alphabetical letters
+    // only or alphabetical letters with one ellisis
+    const std::regex subscript_regex("[A-Za-z]*|[A-Za-z]*\\.\\.\\.[A-Za-z]*");
+
+    // create regular expression to match ellipsis label
+    const std::regex ellipsis_regex("\\.\\.\\.");
+
+    // remove extra white-spaces from the equation
+    m_equation.erase(std::remove_if(m_equation.begin(), m_equation.end(), ::isspace),
+                     m_equation.end());
+
+    // split equation to input subscripts and an output subscript
+    auto pos_output_delimeter = m_equation.find("->");
+    auto input_subscripts_str = m_equation.substr(0, pos_output_delimeter);
+
+    // split the input subscripts into a vector of input subscripts
+    bool is_ellipsis_met = false;
+    std::smatch ellipsis_match;
+    input_subscripts.clear();
+    std::istringstream input;
+    input.str(input_subscripts_str);
+    for (std::string input_subscript; std::getline(input, input_subscript, ',');)
+    {
+        // check that input subscript contains only alphabetic letter or ellipsis
+        if (std::regex_match(input_subscript, subscript_regex) == false)
```

Comment:
Let's convert all the checks in this function which results in `return false;` statement into the `NODE_VALIDATION_CHECK` call. Because what you do here is exactly validating the node attributes. So we can make this function to return `void`.

Reply:
Fixed.